Lib/glyphsLib/parser.py

```diff
@@ -114,17 +114,113 @@ def _parse_list(self, text, i):
         i += len(parsed)
         return res, i
 
+    # glyphs only supports octal escapes between \000 and \077
+    _unescape_re = re.compile(ur'(\\0[0-7]{2})|(\\U[0-9a-fA-F]{4,6})')
+
+    @staticmethod
+    def _unescape_fn(m):
+        if m.group(1):
+            return unichr(int(m.group(1)[1:], 8))
+        return unichr(int(m.group(2)[2:], 16))
+
     def _trim_value(self, value):
         """Trim double quotes off the ends of a value, un-escaping inner
         double quotes.
+        Also convert escapes to unicode.
         """
 
         if value[0] == '"':
             assert value[-1] == '"'
-            return value[1:-1].replace('\\"', '"')
-        return value
+            value = value[1:-1].replace('\\"', '"')
+        return Parser._unescape_re.sub(Parser._unescape_fn, value)
 
     def _fail(self, message, text, i):
         """Raise an exception with given message and text at i."""
 
         raise ValueError('%s:\n%s' % (message, text[i:i + 79]))
+
+
+class Writer(object):
+    """Write parsed data back to flat file.  Normalizes quoting
+    and indentation."""
+
+    # ints and floats are unquoted, as are strings of letters/digits and
+    # underscore with period or leading forward slash.  Everything else
+    # is quoted.
+    _sym_re = re.compile(
+        r'^(?:-?[0-9]+\.?[0-9]*|[_a-zA-Z0-9/\.][_a-zA-Z0-9\.]*)$')
```

Comment:
Do we need to worry about floats without digits before the dot? Glyphs will read those fine, though I don't know if it will ever generate them.